src/services/measure/measurearea.js

```diff
@@ -0,0 +1,82 @@
+goog.provide('ngeo.interaction.MeasureArea');
+
+goog.require('ngeo.interaction.Measure');
+goog.require('ol.geom.Polygon');
+goog.require('ol.interaction.Draw');
+
+
+
+/**
+ * @classdesc
+ * Interaction dedicated to measure length.
+ *
+ * @constructor
+ * @extends {ngeo.interaction.Measure}
+ * @export
+ */
+ngeo.interaction.MeasureArea = function() {
+
+  goog.base(this);
+
+  this.drawInteraction = new ol.interaction.Draw(
+      /** @type {olx.interaction.DrawOptions} */ ({
+        type: 'Polygon',
+        features: this.overlay.getFeatures()
+      }));
+
+
+  /**
+   * Message to show when the user is drawing a line.
+   * @type {string}
+   * @private
+   */
+  this.continueMsg_ = 'Click to continue drawing the polygon';
+
+
+  /**
+   * Message to show when the user is drawing a polygon.
+   * @type {string}
+   * @private
+   */
+  this.finishMsg_ = 'Double-click or click starting point to finish';
+
+};
+goog.inherits(ngeo.interaction.MeasureArea, ngeo.interaction.Measure);
+
+
+/**
+ * @inheritDoc
+ */
+ngeo.interaction.MeasureArea.prototype.handleMeasure = function(callback) {
+  var geom = /** @type {ol.geom.Polygon} */
+      (this.sketchFeature.getGeometry());
+  var output = this.formatMeasure_(geom);
+  var verticesCount = geom.getCoordinates()[0].length;
+  var helpMsg = this.continueMsg_;
+  var coord = null;
+  if (verticesCount > 2) {
+    helpMsg = this.finishMsg_;
+    coord = geom.getInteriorPoint().getCoordinates();
+  }
+  callback(output, coord, helpMsg);
+};
+
+
+/**
+ * Format measure output.
+ * @param {ol.geom.Polygon} polygon
+ * @return {string}
+ * @private
+ */
+ngeo.interaction.MeasureArea.prototype.formatMeasure_ = function(polygon) {
+  var area = polygon.getArea();
```

Comment:
Do you want to add an option to do geodesic calculations? See https://github.com/openlayers/ol3/pull/3222/commits